lib/spack/llnl/util/filesystem.py

```diff
@@ -1233,15 +1234,17 @@ def find_system_libraries(libraries, shared=True):
 
     for library in libraries:
         for root in search_locations:
-            result = find_libraries(library, root, shared, recursive=True)
+            result = find_libraries(library, root, shared, recursive=True,
+                                    return_empty=True)
             if result:
                 libraries_found += result
                 break
 
     return libraries_found
 
 
-def find_libraries(libraries, root, shared=True, recursive=False):
+def find_libraries(libraries, root, shared=True, recursive=False,
+                   return_empty=False):
```

Comment:
@alalazo I hope you don't mind with this extension of `find_libraries` which seems to work as far as I can tell and tested on various packages.